xclim/sdba/_adjustment.py

```diff
@@ -615,3 +615,323 @@ def extremes_adjust(
+    gridX, muX, idx_binX = u.histogram(X, bin_width, bin_origin)
+    gridY, muY, _ = u.histogram(Y, bin_width, bin_origin)
+
+    # Compute the optimal transportation plan
+    plan = u.optimal_transport(gridX, gridY, muX, muY, num_iter_max)
+
+    # Get the source positions expressed in terms of bin indices
+    idx_gridX = np.floor((gridX - bin_origin) / bin_width)
+
+    # regroup the indices of all the points belonging to a same bin
+    idx_sort = np.lexsort(idx_binX[:, ::-1].T)
+    sorted_bins = idx_binX[idx_sort]
+    _, idx_start, idx_count = np.unique(
+        sorted_bins, return_index=True, return_counts=True, axis=0
+    )
+    idx_start = np.sort(idx_start)
+    bin_groups = np.split(idx_sort, idx_start[1:])
+
+    out = np.empty(X.shape)
+    rng = np.random.default_rng()
+    # The plan row corresponding to a source bin indicates its probabilities to be transported to every target bin
+    for i, bin_group in enumerate(bin_groups):
+        # Get the plan row of this bin
+        pi = np.where((idx_binX[bin_group[0]] == idx_gridX).all(1))[0][0]
+        # Pick as much target bins for this source bin as there are points in the source bin
+        choice = rng.choice(range(muY.size), p=plan[pi, :], size=idx_count[i])
+        out[bin_group] = gridY[choice]
+
+    if spray_bins:
+        if isinstance(bin_width, list):
+            bin_width = np.array(bin_width)
+        out += np.random.uniform(low=-bin_width / 2, high=bin_width / 2, size=out.shape)
+
+    return out
+
+
+@map_groups(scen=[Grouper.DIM])
+def otc_adjust(
+    ds: xr.Dataset,
+    dim: list,
+    bin_width: list | None = None,
+    bin_origin: list | None = None,
+    num_iter_max: int | None = 100_000_000,
+    spray_bins: bool = True,
+):
+    """Optimal Transport Correction of the bias of `hist` with respect to `ref`.
+
+    Notes
+    -----
+    Dataset variables:
+      ref : training target
+      hist : training data
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        The dataset containing the data.
+    dim : list
+        The dimensions that are not variables.
+    bin_width : list | None
+        Bin widths for all dimensions.
+    bin_origin : list | None
+        Bin origins for all dimensions.
+    num_iter_max : int | None
+        Maximum number of iterations used in the earth mover distance algorithm.
+
+    Returns
+    -------
+    xr.Dataset
+        Adjusted data
+    """
+    hist = ds.hist.stack(dim_hist=dim)
+    ref = ds.ref.stack(dim_ref=dim, create_index=False)
+    hist = hist.dropna(dim="dim_hist")
+    ref = ref.dropna(dim="dim_ref")
+
+    scen = xr.apply_ufunc(
+        _otc_adjust,
+        hist,
+        ref,
+        kwargs=dict(
+            bin_width=bin_width,
+            bin_origin=bin_origin,
+            num_iter_max=num_iter_max,
+            spray_bins=spray_bins,
+        ),
+        input_core_dims=[["dim_hist", "multivar"], ["dim_ref", "multivar"]],
+        output_core_dims=[["dim_hist", "multivar"]],
```

Comment:
following the change above, I think `multivar` should not be hardcode, 
`"multivar"` -> `dim_multivariate`